simpeg/electromagnetics/time_domain/receivers.py

```diff
@@ -145,6 +159,7 @@ def getP(self, mesh, time_mesh, f):
 
         Ps = self.getSpatialP(mesh, f)
         Pt = self.getTimeP(time_mesh, f)
+        Pt = Pt[:, self.active_times(Pt)]
```

Comment:
The one issue I see by adding this line in `getP` is that any time we want to compute the dot product between `P` and any vector, the vector should also be "reduced" to only the active times. This makes the `getP` method less _standalone_: we need to call `active_times` again if we want to compute dot products of `P` with any vector.

Would it be possible to apply the optimization directly in `eval` and `evalDeriv`, without modifying the output of `getP`? I guess I'm asking if filtering out non-active times in the `P` matrix (`P = sp.kron(Pt, Ps)`) is possible and would achieve same performance as doing so in `Pt`.